src/debugger/cli-debugger-scripting.c

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2013-2016 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include <mgba/core/scripting.h>
+#include <mgba-util/string.h>
+#include <mgba-util/vfs.h>
+
+#include <mgba/debugger/debugger.h>
+#include <mgba/internal/debugger/cli-debugger.h>
+
+static const char* CLIScriptEngineName(struct mScriptEngine*);
+static bool CLIScriptEngineInit(struct mScriptEngine*, struct mScriptBridge*);
+static void CLIScriptEngineDeinit(struct mScriptEngine*);
+static bool CLIScriptEngineIsScript(struct mScriptEngine*, const char* name, struct VFile* vf);
+static bool CLIScriptEngineLoadScript(struct mScriptEngine*, const char* name, struct VFile* vf);
+static void CLIScriptEngineRun(struct mScriptEngine*);
+static bool CLIScriptEngineLookupSymbol(struct mScriptEngine*, const char* name, int32_t* out);
+static void CLIScriptDebuggerEntered(struct mScriptEngine*, enum mDebuggerEntryReason, struct mDebuggerEntryInfo*);
+
+struct CLIScriptStatement {
+	char* command;
+	size_t commandLen;
+	struct CLIScriptStatement* next;
+};
+
+struct CLIScriptEngine {
+	struct mScriptEngine d;
+	struct mScriptBridge* sb;
+	struct CLIScriptStatement* script;
+};
+
+static void CLIScriptClear(struct CLIScriptEngine* engine) {
+	while (engine->script) {
+		struct CLIScriptStatement* statement = engine->script;
+		free(statement->command);
+		engine->script = statement->next;
+		free(statement);
+	}
+}
+
+struct CLIScriptEngine* CLICreateScriptEngine(void) {
+	struct CLIScriptEngine* engine = malloc(sizeof(*engine));
+	engine->d.name = CLIScriptEngineName;
+	engine->d.init = CLIScriptEngineInit;
+	engine->d.deinit = CLIScriptEngineDeinit;
+	engine->d.isScript = CLIScriptEngineIsScript;
+	engine->d.loadScript = CLIScriptEngineLoadScript;
+	engine->d.run = CLIScriptEngineRun;
+	engine->d.lookupSymbol = CLIScriptEngineLookupSymbol;
+	engine->d.debuggerEntered = CLIScriptDebuggerEntered;
+	engine->sb = NULL;
+	engine->script = NULL;
+	return engine;
+}
+
+void CLIDebuggerScriptEngineInstall(struct mScriptBridge* sb) {
+	struct CLIScriptEngine* se = CLICreateScriptEngine();
+	mScriptBridgeInstallEngine(sb, &se->d);
+}
+
+const char* CLIScriptEngineName(struct mScriptEngine* se) {
+	UNUSED(se);
+	return "cli-debugger";
+}
+
+bool CLIScriptEngineInit(struct mScriptEngine* se, struct mScriptBridge* sb) {
+	struct CLIScriptEngine* engine = (struct CLIScriptEngine*) se;
+	engine->sb = sb;
+	return true;
+}
+
+void CLIScriptEngineDeinit(struct mScriptEngine* se) {
+	struct CLIScriptEngine* engine = (struct CLIScriptEngine*) se;
+	CLIScriptClear(engine);
+	free(se);
+}
+
+bool CLIScriptEngineIsScript(struct mScriptEngine* se, const char* name, struct VFile* vf) {
+	UNUSED(se);
+	UNUSED(vf);
+	return endswith(name, ".mgba");
```

Comment:
Not sure this extension makes sense, given that...medusa exists.

Reply:
Oh right. That was a placeholder until I could talk to you about that. Suggestions?

Reply:
`.mds` for "m(GBA|edusa) debugger script"?

Reply:
Maybe `.mrc` for "m(GBA|edusa) run commands"? Not entirely sure.

Reply:
... so _that's_ what rc stands for. I always thought it meant resource!

Sure. Sounds good to me.